app/controllers/schedule_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class ScheduleController < ApplicationController
+  def save
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    @edition = @document.current_edition
+    current_revision = @edition.revision
+    date_issues = Requirements::DatetimeChecker.new(date_params).date_issues
+    datetime_issues = Requirements::DatetimeChecker.new(datetime_params).datetime_issues
+
+    if date_issues.any? || datetime_issues.any?
+      flash[:day_issues] = date_issues.items_for(:scheduled_day)
+      flash[:month_issues] = date_issues.items_for(:scheduled_month)
+      flash[:year_issues] = date_issues.items_for(:scheduled_year)
+      flash[:datetime_issues] = datetime_issues.items_for(:scheduled_datetime)
+
+      render "documents/show"
+      return
+    end
+
+    new_revision = current_revision.build_revision_update(
+      { scheduled_datetime: format_datetime },
+      current_user,
+    )
+    @edition.assign_revision(new_revision, current_user).save!
+
+    redirect_to document_path(@document)
+  end
+
+private
+
+  def format_datetime
```

Comment:
Perhaps `parse` would be more accurate here ;-).

Reply:
This has now been moved to the `ScheduledDatetimeChecker` as `#parsed_datetime`.